lib/pbench/server/auth/__init__.py

```diff
@@ -62,7 +62,10 @@ def __init__(
         self.headers = CaseInsensitiveDict([] if headers is None else headers)
         self.verify = verify
         self.connection = requests.session()
+        # Note: At this point the OpenIDClient object variables initialization
+        # is done, and we start initializing the class variables.
```

Comment:
OK, I have to ask:  why is an _instance_ initialization setting _class_ attributes?...that seems several kinds of questionable.

If we create a second instance, then it will re-write the class attributes...hopefully, it won't change them from what the first instance set them too, because that would be a recipe for sorrow.

Should the class attributes instead be instance attributes?  Should we be creating a singleton class instance here?

Or, is this another aspect of the "static class" thing which came up in #3056?

Reply:
You raised a good point, instance initialization should not set the class attributes.

On the other hand, these class attributes (`USERINFO_ENDPOINT`, `TOKENINFO_ENDPOINT`, `PUBLIC_KEY`) rely upon the instance attribute that is `server_url` because without making the first request to the authentication server these attributes can not be set. So I guess changing them to instance attributes would make more sense maybe.

Reply:
So, if the class attribute values depend on instance values, then they probably aren't really class attributes.  More likely, what we're looking at is a "Singleton" -- a class which only ever has one instance.  A singleton class usually is created by a factory method, but in Python you can override the `__new__()` method so that multiple attempts to create an instance always return the same one.

And then, we would reference the instance for these attribute values.

Reply:
I addressed it for the `public_key` but the rest are not related to the changes proposed here, so I'll defer updating them in #3056 or separate PR.